src/main/java/com/spotify/annoy/ANNIndex.java

```diff
@@ -65,80 +81,84 @@ private void load(final String filename) throws IOException {
         break;
       }
     }
+  }
 
-    System.err.printf("%s: found %d roots with degree %d\n",
-                      filename, roots.size(), m);
+  private float[] getNodeVector(final int nodeOffset) {
+    float[] v = new float[dimension];
+    for (int i = 0; i < dimension; i++)
+      v[i] = annBuf.getFloat(nodeOffset + kNodeHeaderSize + i * FLOAT_SIZE);
+    return v;
   }
 
-  @Override
-  public final void getNodeVector(final int node, final float[] v) {
```

Comment:
I did it this way originally to avoid generating garbage, at the expense of a clean API...